examples/animation.rs

```diff
@@ -31,14 +31,24 @@ fn main() {
         elwt.set_control_flow(ControlFlow::Poll);
 
         match event {
+            Event::WindowEvent {
+                window_id,
+                event: WindowEvent::Resized(size),
+            } if window_id == window.id() => {
+                if let (Some(width), Some(height)) =
+                    { (NonZeroU32::new(size.width), NonZeroU32::new(size.height)) }
+                {
+                    surface.resize(width, height).unwrap();
```

Comment:
Yeah, we probably should, but perhaps we should do that within Winit?

Actually, why are we using `NonZeroU32` again? Why can't the buffer's dimensions be zero? Do some backends panic? What if the surface is a subsurface of another surface, can it be zero then?

> minimizing

I wouldn't expect a `SurfaceResized` event on minimizing, rather something like `Occluded(true)`.

Reply:
> I wouldn't expect a `SurfaceResized` event on minimizing, rather something like `Occluded(true)`.

On `winit 0.29` we still see this on Windows (and maybe on X11/Wayland too, though I've only used a tiling WM/compositor without minimalization).

> Actually, why are we using `NonZeroU32` again? Why can't the buffer's dimensions be zero? Do some backends panic? What if the surface is a subsurface of another surface, can it be zero then?

I'd assume to skip rendering entirely? Not sure if it's safe/valid to call that on `.resize()`.

Reply:
> we still see this on Windows

Yeah, seems to be a long-standing bug, see https://github.com/rust-windowing/winit/issues/783 and https://github.com/rust-windowing/winit/issues/2015.

> I'd assume to skip rendering entirely?

Hmm, yeah, though Softbuffer _could_ also just handle that internally on the platforms where rendering to a zero-sized buffer would fail, no?

Reply:
Discussion moved to https://github.com/rust-windowing/softbuffer/issues/238